src/net/metaplaylist/index.ts

```diff
@@ -0,0 +1,312 @@
+      ) : IManifestParserObservable {
+
+        const documents = response.responseData;
+        const manifestsInfos =  documents.manifests;
+
+        const parsedManifestsInfo = manifestsInfos.map((manifestInfos) => {
+          const transport = transports[manifestInfos.transport];
+          const parser = transport.manifest.parser;
+          const fakeResponse = {
+            responseData: manifestInfos.manifest,
+          };
+          return parser({ response: fakeResponse, url: manifestInfos.url })
+          .map((mpd) => {
+            const type = mpd.manifest.type;
+            if (type !== "static") {
+              throw new Error("Content from metaplaylist is not static.");
+            }
+            return {
+              manifest: mpd.manifest,
+              transport: manifestInfos.transport,
+              url: manifestInfos.url,
+              startTime: manifestInfos.startTime,
+              endTime: manifestInfos.endTime,
+            };
+          });
+        });
+
+        return Observable.combineLatest(parsedManifestsInfo).map((contents) => {
+            const manifest = generateManifest(contents, url);
+            return {
+              manifest,
+              url,
+            };
+        });
+      },
+    };
+
+    const segmentPipeline = {
+      loader({
+        adaptation,
+        init,
+        manifest,
+        period,
+        representation,
+        segment,
+      } : ISegmentLoaderArguments) : ILoaderObservable<Uint8Array|ArrayBuffer> {
+        if (!segment.privateInfos) {
+          throw new Error("Segments from metaplaylist must have private infos.");
+        }
+        const transportType = getTypeFromPrivateInfos(segment.privateInfos);
+        const transport = transports[transportType];
+        const segmentLoader = transport.video.loader;
+        return segmentLoader({
+          adaptation,
+          init,
+          manifest,
+          period,
+          representation,
+          segment,
+        });
+      },
+
+      parser(args : ISegmentParserArguments<Uint8Array|ArrayBuffer>
+        ) : SegmentParserObservable {
+          const { period, init, segment } = args;
+          if (!segment.privateInfos) {
+            throw new Error("Segments from metaplaylist must have private infos.");
+          }
+          const transportType = getTypeFromPrivateInfos(segment.privateInfos);
+          const transport = transports[transportType];
+
+          if (!transport) {
+            throw new Error("Segments from metaplaylist must have private infos.");
+          }
+          const segmentParser = transport.video.parser;
+
+          return segmentParser(args).map(({ segmentData, segmentInfos }) => {
+            let segmentPatchedData;
+            const offset = (period.start || 0) *
+              (init ? (init.timescale || segment.timescale) :  segment.timescale);
+
+            if(segmentData !== undefined){
```

Comment:
spacing (if looks like a function here, we're doing TypeScript, we do not need such functional shenanigans)